File: Python/Day199/subarraySum.py
arr = [1, -2, 6, -1, 3]

def maxAndMinSum(arr): 
    max = 0
    min = 100000
    for i in range(len(arr)): 
        for j in range(i, len(arr)):
            sum = 0
            for k in range(i, j+1):
                sum += arr[k]
            print(sum)
            if(sum > max):
                max = sum
            if(sum < min):
                min = sum
            
    print("Maximum sum : ", max) 
    print("Minimum sum : ", min)

maxAndMinSum(arr)

File: Java/Day199/SubarraySum.java
package Java.Day199;

public class SubarraySum {
    public static void maxAndMinSum(int[] arr) {
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;
        int s2;
        for (int i = 0; i < arr.length; i++) {
            for (int j = i; j < arr.length; j++) {
                s2 = 0;
                for (int k = i; k <=j; k++){
                    s2 += arr[k];
                }
                System.out.println(s2);
                if(s2 > max) {
                    max = s2;
                }
                if(s2 < min) {
                    min = s2;
                }
            }
        }
        System.out.println("Maximum subarray sum : " + max);
        System.out.println("Minimum subarray sum : " + min);
    }
     public static void main(String args[]) {
        int[] arr = {1, -2, 6, -1, 3};
        maxAndMinSum(arr);
    }
}
